test/libsolidity/syntaxTests/specialFunctions/encodeCall_fail_funType.sol

```diff
@@ -0,0 +1,72 @@
+interface I {
+	function fExternal(uint256 p, string memory t) external;
+}
+
+contract Other {
+	function fExternal(uint) external pure {}
+	function fPublic(uint) public pure {}
+	function fInternal(uint) internal pure {}
+}
+
+library L {
+	function fExternal(uint256 p, string memory t) external {}
+	function fInternal(uint256 p, string memory t) internal {}
+}
+
+contract Base {
+	function baseFunction(uint) internal pure {}
+	function baseFunctionPublic(uint) public pure {}
+}
+
+contract C is Base {
+	using L for uint256;
+
+	function f(int a) public {}
+	function f2(int a, string memory b) public {}
+	function f3(int a, int b) public {}
+	function f4() public {}
+	function fInternal(uint256 p, string memory t) internal {}
+
+	function failFunctionPtrMissing() public returns(bytes memory) {
+		return abi.encodeCall(1, this.f);
+	}
+	function failFunctionPtrWrongType() public returns(bytes memory) {
+		return abi.encodeCall(abi.encodeCall, (1, 2, 3, "test"));
+	}
+	function failFunctionInternal() public returns(bytes memory) {
+		return abi.encodeCall(fInternal, (1, "123"));
+	}
+	function failFunctionInternalFromVariable() public returns(bytes memory) {
+		function(uint256, string memory) internal localFunctionPointer = fInternal;
+		return abi.encodeCall(localFunctionPointer, (1, "123"));
+	}
+	function failLibraryPointerCall() public {
+		abi.encodeCall(L.fInternal, (1, "123"));
+		abi.encodeCall(L.fExternal, (1, "123"));
+	}
+	function failBoundLibraryPointerCall() public returns (bytes memory) {
+		uint256 x = 1;
+		return abi.encodeCall(x.fExternal, (1, "123"));
+	}
+	function viaBaseDeclaration() public pure returns (bytes memory) {
+		return abi.encodeCall(C.f, (2));
+	}
+	function viaBaseDeclaration() public pure returns (bytes memory) {
+		return bytes.concat(
+			abi.encodeCall(Base.baseFunctionPublic, (1)),
+			abi.encodeCall(Base.baseFunction, (1))
+		);
+	}
```

Comment:
Both functions are called `viaBaseDeclaration()`. That's an error (and not one we're supposed to test here :)).